.github/workflows/ci.yml

```diff
@@ -106,13 +145,33 @@ jobs:
         if: steps.spack-config-update.conclusion == 'success' || steps.spack-update.conclusion == 'success'
         run: ln --symbolic --relative --verbose --force ${SPACK_CONFIG_DIR}/* ${SPACK_ROOT}/etc/spack/
 
-      - name: Checkout ${{ github.repository }}
+      - name: Spack - Enable
+        run: . $SPACK_CONFIG_DIR/ci-spack-enable.bash
+
+      - name: Manifest - Checkout ${{ github.repository }}
+        id: checkout
         uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref }}
 
-      - name: Enable ci version of spack
-        run: . $SPACK_CONFIG_DIR/ci-spack-enable.bash
+      - name: Manifest - Install Compilers Locally
+        # Some compilers may not be available upstream, so we install them locally in a separate spack.yaml
+        if: inputs.spack-compiler-manifest-path != ''
+        run: |
+          spack env activate compilers --create --prompt --envfile ${{ inputs.spack-compiler-manifest-path }}
+          spack --debug install --fail-fast
+          spack env deactivate
+          yq '.spack.specs[]' ${{ inputs.spack-compiler-manifest-path }} | while read compiler; do
+            spack load $compiler
+          done
+          spack compiler find
+          spack compiler list
+
+      - name: Manifest - Replace usages of __REF__ with ${{ inputs.ref }}
+        # This means that the caller manifest files can remain unchanged, but still vary based on the ref
+        run: sed -i "s/__REF__/${{ inputs.ref }}/g" ${{ inputs.spack-manifest-path }}
```

Comment:
So the manifest is a template with `__REF__` replaced dynamically?

I think we might need to have a dialogue about future requirements. Should we standardise on a template approach, e.g. [jinja](https://jinja.palletsprojects.com/en/stable/)?